### PR TITLE
[FIX] sale_product_configurator: avoid returning archived variant

### PR DESCRIPTION
**Pull request by livaios**
If product has no_variant attributes with 2 values and the first value was deleted, 
it is still returning it in the product configurator. 
Filter the template's value ids to return only active variants 
opw-3570604

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by livaios** [comment]
Hello @chevalierv 

This is a different issue
What I was handling is that:

- create an attribute "Radio, no variants" with two custom values;
- create so with that product
- delete first custom value
- create another so with the same product
- it fails when you try and edit the product on the 2nd so

File: addons/sale_product_configurator/controllers/main.py [hunk 1]
```diff
@@ -47,10 +47,10 @@ def get_product_configurator_values(
             combination = request.env['product.template.attribute.value'].browse(ptav_ids).filtered(
                 lambda ptav: ptav.product_tmpl_id.id == product_template_id
             )
-            # Set attributes that are in mode `no_variant` and have only one value
+            # Set attributes that are in mode `no_variant`
             unconfigured_ptals = product_template.attribute_line_ids - combination.attribute_line_id
-            combination += unconfigured_ptals.filtered(lambda ptal: ptal.value_count == 1).mapped(
-                lambda ptal: ptal.product_template_value_ids[:1]
+            combination += unconfigured_ptals.mapped(
+                lambda ptal: ptal.product_template_value_ids._only_active()[:1]
```
Comment (chevalierv) [inline addons/sale_product_configurator/controllers/main.py:53]: @vchu-odoo Do you remember why we needed the `filtered` ?

Comment (vchu-odoo) [reply addons/sale_product_configurator/controllers/main.py:53]: The problem that we fixed that if the attribute had only one value it was not included in the combination, hence was not included in exclusions. Turned out that it was not only the case for one value, @Feyensv suggested to remove filter.

File: addons/sale_product_configurator/controllers/main.py [hunk 2]
```diff
@@ -47,10 +47,10 @@ def get_product_configurator_values(
             combination = request.env['product.template.attribute.value'].browse(ptav_ids).filtered(
                 lambda ptav: ptav.product_tmpl_id.id == product_template_id
             )
-            # Set attributes that are in mode `no_variant` and have only one value
+            # Set attributes that are in mode `no_variant`
             unconfigured_ptals = product_template.attribute_line_ids - combination.attribute_line_id
```
Comment (Feyensv) [inline addons/sale_product_configurator/controllers/main.py:51]:
```suggestion
            unconfigured_ptals = (product_template.attribute_line_ids - combination.attribute_line_id).filtered(lambda ptal.attribute.type != 'multi')
```
in 17.0 fw-port

File: addons/sale_product_configurator/controllers/main.py [hunk 3]
```diff
@@ -47,10 +47,10 @@ def get_product_configurator_values(
             combination = request.env['product.template.attribute.value'].browse(ptav_ids).filtered(
                 lambda ptav: ptav.product_tmpl_id.id == product_template_id
             )
-            # Set attributes that are in mode `no_variant` and have only one value
+            # Set attributes that are in mode `no_variant`
```
Comment (Feyensv) [inline addons/sale_product_configurator/controllers/main.py:50]:
```suggestion
            # Set missing attributes (unsaved no_variant attributes, or new attribute on existing product)
```

**Review by chevalierv — APPROVED** [review]
Hello @livaios 

Thanks for your work on this!

@robodoo r+

**Pull request closed without merging**